client/src/main/scala/org/http4s/client/middleware/Retry.scala

```diff
@@ -61,8 +47,47 @@ object Retry {
   }
 }
 
-
 object RetryPolicy {
+  /** Decomposes a retry policy into components that are typically configured
+    * individually.
+    * 
+    * @param backoff a function of attempts to an optional
+    * FiniteDuration.  Return None to stop retrying, or some
+    * duration after which the request will be retried.  See
+    * `exponentialBackoff` for a useful implementation.
+    * 
+    * @param retriable determines whether the request is retriable
+    * from the request and either the throwable or response that was
+    * returned.  Defaults to `defaultRetriable`.
+    */
+  def apply(
+    backoff: Int => Option[FiniteDuration],
+    retriable: (Request, Either[Throwable, Response]) => Boolean = defaultRetriable
+  ): RetryPolicy = { (req, result, retries) =>
+    if (retriable(req, result)) backoff(retries)
+    else None
+  }
+
+  /** Statuses that are retriable, per HTTP spec */
+  val RetriableStatuses = Set(
+    RequestTimeout,
+    // TODO Leaving PayloadTooLarge out until we model Retry-After
+    InternalServerError,
+    ServiceUnavailable,
+    BadGateway,
+    GatewayTimeout
+  )
+
+  /** Default logic for whether a request is retriable. */
+  def defaultRetriable(req: Request, result: Either[Throwable, Response]): Boolean = {
+    if (req.isIdempotent)
```

Comment:
This method doesn't exist in fs2.  The best we can do is check that the request method is idempotent.   We need to decide whether to default to retrying all idempotent methods, or only methods known not to have a body.  We could conceivably check a content length of zero, but even that doesn't mean the body launches the nukes.